ocpmodels/trainers/ocp_trainer.py

```diff
@@ -514,20 +514,30 @@ def predict(
                 return predictions
 
             ### Get unique system identifiers
-            sids = batch.sid.tolist()
+            sids = list(batch.sid)
             ## Support naming structure for OC20 S2EF
             if "fid" in batch:
-                fids = batch.fid.tolist()
+                fids = list(batch.fid)
                 systemids = [f"{sid}_{fid}" for sid, fid in zip(sids, fids)]
             else:
                 systemids = [f"{sid}" for sid in sids]
 
             predictions["ids"].extend(systemids)
 
         for key in predictions:
-            predictions[key] = np.array(predictions[key])
-
-        self.save_results(predictions, results_file)
+            # allow for lists of 'zero dim' arrays
+            predictions[key] = np.vstack(predictions[key]).squeeze()
+
+        if self.is_debug:
```

Comment:
Is there a scenario where `save_results()` gets called under debug mode? We should just raise an error at the start of prediction if in debug mode and not wait for the writing step.

Reply:
Pinging @emsunshine for this